examples/run/run.cpp

```diff
@@ -729,11 +729,12 @@ static int apply_chat_template(LlamaData & llama_data, const bool append) {
 
 // Function to tokenize the prompt
 static int tokenize_prompt(const llama_vocab * vocab, const std::string & prompt,
-                           std::vector<llama_token> & prompt_tokens) {
-    const int n_prompt_tokens = -llama_tokenize(vocab, prompt.c_str(), prompt.size(), NULL, 0, true, true);
+                           std::vector<llama_token> & prompt_tokens, const LlamaData & llama_data,
+                           const bool is_first) {
+    const int n_prompt_tokens = -llama_tokenize(vocab, prompt.c_str(), prompt.size(), NULL, 0, is_first, true);
     prompt_tokens.resize(n_prompt_tokens);
-    if (llama_tokenize(vocab, prompt.c_str(), prompt.size(), prompt_tokens.data(), prompt_tokens.size(), true,
-                       true) < 0) {
+    if (llama_tokenize(vocab, prompt.c_str(), prompt.size(), prompt_tokens.data(), prompt_tokens.size(),
+                       llama_get_kv_cache_used_cells(llama_data.context.get()) == 0, true) < 0) {
```

Comment:
```suggestion
    if (llama_tokenize(vocab, prompt.c_str(), prompt.size(), prompt_tokens.data(), prompt_tokens.size(),
                       is_first, true) < 0) {
```

Reply:
Ah, I didn't even notice. Maybe this way it is better than introducing the `is_token` parameter.

Reply:
I'm not pretending I know the details and just copying simple-chat :)

Reply:
I pushed this variant to both `simple-chat` and `run` - seems much simpler than the `is_first` solution: https://github.com/ggerganov/llama.cpp/pull/11311